engine/src/main/java/org/hibernate/search/engine/spi/EntityIndexBinding.java

```diff
@@ -50,6 +50,18 @@
 	void postInitialize(Set<Class<?>> indexedClasses);
 
 	/**
+	 * @return the {@code IndexManager} impl used by this entity type
+	 */
+	Class<? extends IndexManager> getIndexManagerImpl();
+
+	/**
+	 * @param indexManagerCandidate the tested {@code IndexManager} implementation
+	 *
+	 * @return whether this entity is managed by the given {@code IndexManager} impl
+	 */
+	boolean isManagedBy(Class<? extends IndexManager> indexManagerCandidate);
```

Comment:
I agree it can be avoided but I think it's something that should be available in EntityIndexBinding. That's why I added it there.

Probably better to wait for @Sanne for the SPI changes as I'm pretty sure he will have a strong opinion about them.